components/script/dom/webgl2renderingcontext.rs

```diff
@@ -180,7 +181,7 @@ impl WebGL2RenderingContext {
     #[cfg_attr(crown, allow(crown::unrooted_must_root))]
     pub(crate) fn new(
         window: &Window,
-        canvas: &HTMLCanvasElementOrOffscreenCanvas,
+        canvas: HTMLCanvasElementOrOffscreenCanvas,
```

Comment:
I think we should actually use rooted here as arg as alloc (of reflector) can cause GC.

Reply:
Of course, this will be unsafe for potential future use with the new API, and
the engine developer should understand this and adapt the code accordingly.

Anyway, does your original comment apply only to WebGL2 or also to any other context types?

Reply:
Also for webgl, but I did not review others.

Reply:
Done